cJSON.c

```diff
@@ -452,20 +431,22 @@ static cJSON_bool print_number(const cJSON * const item, printbuffer * const out
     {
         length = sprintf((char*)number_buffer, "null");
     }
-    else if ((fabs(floor(d) - d) <= DBL_EPSILON) && (fabs(d) < 1.0e60))
```

Comment:
Update: `modf` can be used to test if it is an integer.

If you don't want to tackle this, I can do this myself after the PR is merged.